### PR TITLE
docs(fundamentals): updates testing to explain new setLogger method

### PR DESCRIPTION
**Pull request by nvdnkpr**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[X] Docs
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: https://github.com/nestjs/nest/pull/6071

## What is the new behavior?
Docs for new `setLogger`method of TestModuleBuilder (https://github.com/nestjs/nest/pull/6071)

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by johannesschobel — COMMENTED** [review]
should we then add some more details on how to properly log things in the `TestModuleBuilder`? because actually now the docs would only state, that you should extend the `Logger` class and tell to use it - but how? ;) @nvdnkpr

**Comment by nvdnkpr** [comment]
@johannesschobel I think the initial intend (behind the built-in TestLogger service) was to have a minimal version that applies to the required interface that must be implemented for the general LoggerService of Nest. My extension that I document with this PR is just the attempt to make this customizable. 
So in order to customize it, a user to understand that he has to write a class that implements the LoggerService. This is also really well described in the big 'Logging' chapter of Nest, to which I created the Link.
I also intentionally linked to the sub-chapter that holds the explicit implementation of the LoggerService interface. 

So, I think with the general knowledge of "how to create a custom Logger" my change will only enable the user to "create the custom Logger and use when creating the NestApplication via TestModuleBuilder.

**Review by jmcdo29 — APPROVED** [review]
If the core PR gets approved and merged, this looks good as well

**Comment by nvdnkpr** [comment]
@kamilmysliwiec any Progress on the Core PR ? 😊

File: content/fundamentals/unit-testing.md [hunk 1]
```diff
@@ -256,6 +256,7 @@ We simulate HTTP tests using the `request()` function from Supertest. We want th
 
 In this example, we also provide an alternate (test-double) implementation of the `CatsService` which simply returns a hard-coded value that we can test for. Use `overrideProvider()` to provide such an alternate implementation. Similarly, Nest provides methods to override guards, interceptors, filters and pipes with the`overrideGuard()`, `overrideInterceptor()`, `overrideFilter()`, and `overridePipe()` methods respectively.
 
+
```
Comment (kamilmysliwiec) [inline content/fundamentals/unit-testing.md:259]:
```suggestion
```

File: content/fundamentals/unit-testing.md [hunk 2]
```diff
@@ -264,6 +264,9 @@ Each of the override methods returns an object with 3 different methods that mir
 
 Each of the override method types, in turn, returns the `TestingModule` instance, and can thus be chained with other methods in the [fluent style](https://en.wikipedia.org/wiki/Fluent_interface). You should use `compile()` at the end of such a chain to cause Nest to instantiate and initialize the module.
 
+While testing you sometimes also want to provide a custom logger e.g. when the tests are run on a CI server. Use `setLogger()` with a `Logger` instance that is derived from the
```
Comment (kamilmysliwiec) [inline content/fundamentals/unit-testing.md:267]:
```suggestion
Also, sometimes you may want to provide a custom logger e.g. when the tests are run (for example, on a CI server). Use the `setLogger()` method and pass an object that fulfills the `LoggerService` interface to instruct the `TestModuleBuilder` how to log during tests (by default, only "error" logs will be logged to the console).
```

File: content/fundamentals/unit-testing.md [hunk 3]
```diff
@@ -264,6 +264,9 @@ Each of the override methods returns an object with 3 different methods that mir
 
 Each of the override method types, in turn, returns the `TestingModule` instance, and can thus be chained with other methods in the [fluent style](https://en.wikipedia.org/wiki/Fluent_interface). You should use `compile()` at the end of such a chain to cause Nest to instantiate and initialize the module.
 
+Also, sometimes you may want to provide a custom logger e.g. when the tests are run (for example, on a CI server). Use the `setLogger()` method and pass an object that fulfills the `LoggerService` interface to instruct the `TestModuleBuilder` how to log during tests (by default, only "error" logs will be logged to the console).
+[Logger class](https://docs.nestjs.com/techniques/logger#injecting-a-custom-logger) to tell the TestModuleBuilder how to log during tests.
```
Comment (kamilmysliwiec) [inline content/fundamentals/unit-testing.md:268]:
```suggestion
```

**Pull request merged**